### PR TITLE
Add Configuration to be able to use client cert authentication

### PR DESCRIPTION
**Pull request by raynay-r**
Hi!

We are evaluating k6 at the moment and want to use this plugin to output to Elasticsearch. Currently this plugin doesn't work for us because our Elasticsearch cluster is configured to only accept client cert authentication. Since that can't be changed at the moment, I made a few changes to support the additional authentication method.

Would be great if you could take a look at the changes and merge them in if everything looks good. If you have any questions or improvements I am available.

Thanks!

BR
raynay

### CODE REVIEW

**Comment by raynay-r** [comment]
I am not able to sign the contributer agreement. When I click sign I just get the following empty white page:
image

**Comment by danielmitterdorfer** [comment]
@raynay-r thanks for your contribution! Regarding the CLA issue, the form to sign the contributor agreement loads for me in Firefox 131.0.2 but it took a couple of seconds (didn't measure but it was maybe 5-10 seconds). I've noticed that the form is loaded via an iframe from https://na3.docusign.net/. Is it possible that this URL is blocked on your end?

**Comment by raynay-r** [comment]
> @raynay-r thanks for your contribution! Regarding the CLA issue, the form to sign the contributor agreement loads for me in Firefox 131.0.2 but it took a couple of seconds (didn't measure but it was maybe 5-10 seconds). I've noticed that the form is loaded via an iframe from https://na3.docusign.net/. Is it possible that this URL is blocked on your end?

Seems I was just not patient enough :D I signed the CLA, do I have to do anything else?

**Comment by danielmitterdorfer** [comment]
@raynay-r it seems that the CLA check did not succeed. The CLA check works based on the email address of your commits and I believe there is a mismatch here. Looking at the git history of this PR the commit author address is `@extern.....at` whereas I could find a CLA signed with your `@gmx.net` address (intentionally not sharing both email addresses publicly but I hope you can still make sense of them). 

So can I please ask you to align the two (either sign the CLA with the `@extern.....at` address or create a new PR with commits from the `@gmx.net` address? Sorry to have you go through this legalese hassle... . But afterwards I'm happy to look at your PR.

**Comment by raynay-r** [comment]
No worries :) Thanks for catching the mistake with the wrong mail address. I rebased the commit to have the correct author and the check succeeds now.

**Review by danielmitterdorfer — COMMENTED** [review]
I have one minor comment but apart from that it looks pretty good.

File: pkg/esoutput/config.go [hunk 1]
```diff
@@ -45,6 +46,9 @@ type Config struct {
 	CACert             null.String `json:"caCertFile" envconfig:"K6_ELASTICSEARCH_CA_CERT_FILE"`
 	InsecureSkipVerify null.Bool   `json:"insecureSkipVerify" envconfig:"K6_ELASTICSEARCH_INSECURE_SKIP_VERIFY"`
 
+	ClientCert null.String `json:"clientCertFile" envconfig:"K6_ELASTICSEARCH_CLIENT_CERT_FILE"`
```
[the remark this answers is not in the document]
Comment (raynay-r) [reply pkg/esoutput/config.go:49]: Done.

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for the PR! Looks good to me; I'll merge it now.

**Pull request merged**